### PR TITLE
Chi 1974 case merging design cleanup

### PR DESCRIPTION
**Pull request by acedeywin**
Primary reviewer:

## Description

 - New feature: This task is part of the QA feedback for case merging design clean up where I implemented the feature that handles zero-result situations during case/contact search.

### Checklist
- [ ] Corresponding issue has been opened
- [ ] New tests added
- [ ] Feature flags added
- [ ] Strings are localized
- [ ] Tested for chat contacts
- [ ] Tested for call contacts

### Related Issues
Fixes [CHI-2424](https://tech-matters.atlassian.net/browse/CHI-2424)

### Verification steps

- Try searching for a contact or case that doesn't exist
- Try adding existing contact to case and search for a case that does not exist

[CHI-2424]: https://tech-matters.atlassian.net/browse/CHI-2424?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: plugin-hrm-form/src/components/search/SearchResults/index.tsx [hunk 1]
```diff
@@ -204,6 +246,35 @@ const SearchResults: React.FC<Props> = ({
...
+        <Text padding="0" fontWeight="700" margin="20px" color="#282a2b">
+          <Template code="SearchResultsIndex-NoCasesFound" type={type} />
+        </Text>
+      </Row>
+
+      <Row>
+        <Text margin="44px" decoration="underline" color="#1976D2" cursor="pointer" onClick={openSearchModal}>
+          <Template code="SearchResultsIndex-SearchAgainForCase" type={type.toLocaleLowerCase()} />
+        </Text>
+        {routing.action && (
```
Comment (stephenhand) [inline plugin-hrm-form/src/components/search/SearchResults/index.tsx:262]: When was this 'Add a new case' requirement added? I thought this was supposed to be a simple visual placeholder

File: plugin-hrm-form/src/states/routing/types.ts [hunk 2]
```diff
@@ -93,6 +93,7 @@ type CaseCoreRoute = {
   route: 'case';
   autoFocus?: boolean;
   isCreating?: boolean;
+  // action?: 'select-case';
```
Comment (stephenhand) [inline plugin-hrm-form/src/states/routing/types.ts:96]: Remove

File: plugin-hrm-form/src/components/case/CaseHome.tsx [hunk 3]
```diff
@@ -108,6 +108,8 @@ const CaseHome: React.FC<Props> = ({
   const { caseForms } = definitionVersion;
   const caseLayouts = definitionVersion.layoutVersion.case;
 
+  const isCreating = routing.route === 'case' && routing.isCreating;
```
Comment (stephenhand) [inline plugin-hrm-form/src/components/case/CaseHome.tsx:111]: This is a good improvement. For future reference you can also put this logic in the mapStateToProps function and only pass 'isCreating' into the component props if that's all you need. 

Deconstructing the state as much as possible prior to passing it into the component it helps keep a clean separation between reading the state and applying the state to the UI. Not saying this code needs to change, just for future reference

**Review by stephenhand — APPROVED** [review]
The changes look good, still not sure about the extra requirenments being added but that's not for this PR review

File: plugin-hrm-form/src/components/search/SearchResults/index.tsx (hunk 1 above)
Comment (acedeywin) [reply plugin-hrm-form/src/components/search/SearchResults/index.tsx:262]: @stephenhand  - It's actually on the [design mock](https://www.figma.com/file/WGbudu60QHTKTJvoMD2Su5/Aselo-Figma-Library?type=design&node-id=383-5103&mode=design&t=gCBbvbw2YKC5NV5s-0).

File: plugin-hrm-form/src/states/routing/types.ts (hunk 2 above)
Comment (acedeywin) [reply plugin-hrm-form/src/states/routing/types.ts]: @stephenhand  - Done

**Pull request merged**